bc/module.cpp

```diff
@@ -447,7 +447,34 @@ Value *ModuleParseContext::get_value(const Vector<uint64_t> &ops, unsigned &inde
 {
 	if (index >= ops.size())
 		return nullptr;
-	return get_value(ops[index++], expected_type);
+
```

Comment:
This feels like it duplicates the existing get_value_and_type?

Reply:
It does duplicate it somewhat. I didn't want to make too much of a change here to minimize the possibility of issues elsewhere so I will defer to your guidance here.

Reply:
Here is the dxil for testing.

[650ff513851cfb0f.zip](https://github.com/HansKristian-Work/dxil-spirv/files/6359651/650ff513851cfb0f.zip)

Reply:
Updated the PR without the duplicate code.